### ISSUE TITLE
can anyone provide an example of spherical harmonics complex packing?

### ISSUE THREAD

**Issue by edwardhartnett** [NOAA-EMC/NCEPLIBS-g2c#189]
Can anyone provide a small, but working, example of the settings needed for spherical harmonics complex packing? Is anyone we know using this?

I am trying to write a test for specpack()/specunpack(), but I don't understand how this packing works - it's quite beyond the sort of math I'm used to. I'm sure I can understand it eventually, but it sure would be helpful if someone who knows it could give me a working example...

See: https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_temp5-51.shtml

I've got the test I'm trying up here: https://github.com/NOAA-EMC/NCEPLIBS-g2c/pull/188

Currently the code (and test) report success, but the data I get out of specunpack does not match the data I put into specpack. Which is sad.

**Comment by BoiVuong-NOAA** [comment]
No, I do not know spherical harmonics complex packing.  I do not know any GRIB2 in PROD that uses spherical harmonics complex packing.

**Comment by Hang-Lei-NOAA** [comment]
When I was developing the interoperable physics driver, I remembered that
the stochastic physics used it in some way. Let me search it.

**Comment by Hang-Lei-NOAA** [comment]
You can get some reference from here:
git clone https://github.com/noaa-psd/stochastic_physics.git

stochy_internal_state_mod.F90 contains the spherical harmonic definitions
and arrays
stochy_patterngenerator.F90;
spectral_transforms.F90; get_stochy_pattern.F90; get_stochy_pattern.F90
 initial, convert it.

**Comment by edwardhartnett** [comment]
@Hang-Lei-NOAA can you try to turn this into a test? I can help if you need it...

**Comment by Hang-Lei-NOAA** [comment]
@Edward Hartnett - NOAA Affiliate ***@***.***>  Okay, let me
explore it this summer, besides the library work for models.

**Comment by edwardhartnett** [comment]
OK, this got dropped, but @Hang-Lei-NOAA if you can obtain a file which uses this packing, we can turn it into a test. But we need a data file...
